roles/download/defaults/main.yml

```diff
@@ -17,6 +20,9 @@ download_localhost: False
 # Always pull images if set to True. Otherwise check by the repo's tag/digest.
 download_always_pull: False
 
+# Use the first kube-master if download_localhost is not set
+download_delegate: "{% if download_localhost|bool %}localhost{% else %}{{groups['kube-master'][0]}}{% endif %}"
```

Comment:
this will always be localhost, as true and false is bool, and filter |bool should be avoided, cause it sometimes not works as expected, before all tasks you should make check for proper variable configuration, and in if you put just: `{% if download_localhost %}localhost{% else %}{{groups['kube-master'][0]}}{% endif %}"`

Reply:
Thanks for the suggestion